package.json

```diff
@@ -10,38 +10,38 @@
     "tslint-config-shopify",
     "shopify"
   ],
-  "main": "base.json",
+  "main": "./base.json",
   "scripts": {
-    "test": "yarn run check",
-    "lint:ts": "tslint -c base.json *.ts --project tsconfig.json --type-check",
-    "lint:js": "eslint . --max-warnings 0",
-    "check": "yarn run lint:js && yarn run lint:ts"
+    "prebuild": "yarn run build:clean",
+    "build:clean": "rimraf build rules base.js react.js untyped.js",
+    "build": "tsc",
+    "predist": "yarn run build",
+    "dist": "cp -R build/** .",
+    "postpublish": "yarn run build:clean",
+    "prepublish": "yarn run dist",
+    "pretest": "yarn run dist",
+    "test": "yarn run lint:ts",
+    "lint": "yarn run dist && yarn run lint:ts",
```

Comment:
The `dist` task creates files in the root dir that could be confusing to newcomers.  I think we should just use `build`'s results for linting.  

That means hardcoding paths with `build` inside them, but that helps to make it clear that the project must build itself before it can be fully linted.